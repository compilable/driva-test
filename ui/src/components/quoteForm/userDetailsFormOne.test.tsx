import React from 'react';
import { render } from '@testing-library/react'
import { shallow} from 'enzyme';
import toJSON from "enzyme-to-json"

import UserDetailsFormOne from "./userDetailsFormOne"

test("has the userDetailsFormOne container loaded", () => {
    const getCustomErrorMessage = jest.fn()
    const processForm = jest.fn()
    const userDetailsFormOne = shallow(<UserDetailsFormOne formName={"q1"} showCustomError={getCustomErrorMessage} processForm={processForm} />);
    expect(toJSON(userDetailsFormOne)).toMatchSnapshot()
});

test("has the userDetailsFormOne title loaded", () => {
    const getCustomErrorMessage = jest.fn()
    const processForm = jest.fn()
    const userDetailsFormOne = render(<UserDetailsFormOne formName={"q1"} showCustomError={getCustomErrorMessage} processForm={processForm} />);
    expect(userDetailsFormOne.getAllByText("Tell us about yourself"))
});

test("has the InputElement loaded", () => {
    const getCustomErrorMessage = jest.fn()
    const processForm = jest.fn()
    const userDetailsFormOne = render(<UserDetailsFormOne formName={"q1"} showCustomError={getCustomErrorMessage} processForm={processForm} />);
    expect(userDetailsFormOne.getAllByRole('textbox').length).toEqual(5)
});

test("has the Button loaded", () => {
    const getCustomErrorMessage = jest.fn()
    const processForm = jest.fn()
    const userDetailsFormOne = render(<UserDetailsFormOne formName={"q1"} showCustomError={getCustomErrorMessage} processForm={processForm} />);
    expect(userDetailsFormOne.getAllByRole('button').length).toEqual(1)
});