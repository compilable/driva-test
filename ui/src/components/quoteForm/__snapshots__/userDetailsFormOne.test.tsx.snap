// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`has the userDetailsFormOne container loaded 1`] = `
<Container
  style={
    Object {
      "margin": "5%",
    }
  }
>
  <Header
    as="h2"
  >
    Tell us about yourself
  </Header>
  <Card
    fluid={true}
  >
    <Form
      as="form"
      style={
        Object {
          "margin": "2%",
        }
      }
    >
      <InputElement
        formErrors={Object {}}
        formField="userFirstName"
        lableText="First Name"
        placeholderText="As it appears on your license"
        required={true}
        showCustomError={[MockFunction]}
        validationHook={
          Object {
            "name": "userFirstName",
            "onBlur": [Function],
            "onChange": [Function],
            "ref": [Function],
          }
        }
      />
      <InputElement
        formField="userMiddleName"
        lableText="Middle Name"
        placeholderText="Optional"
        required={false}
        validationHook={
          Object {
            "name": "userMiddleName",
            "onBlur": [Function],
            "onChange": [Function],
            "ref": [Function],
          }
        }
      />
      <InputElement
        formErrors={Object {}}
        formField="userLastName"
        lableText="Last Name"
        placeholderText="As it appears on your license"
        required={true}
        showCustomError={[MockFunction]}
        validationHook={
          Object {
            "name": "userLastName",
            "onBlur": [Function],
            "onChange": [Function],
            "ref": [Function],
          }
        }
      />
      <InputElement
        formErrors={Object {}}
        formField="userMobileNumber"
        lableText="Mobile Number"
        placeholderText="+61"
        required={true}
        showCustomError={[MockFunction]}
        validationHook={
          Object {
            "name": "userMobileNumber",
            "onBlur": [Function],
            "onChange": [Function],
            "ref": [Function],
          }
        }
      />
      <InputElement
        formErrors={Object {}}
        formField="userEmail"
        lableText="Email"
        placeholderText="Please enter a valid email"
        required={true}
        showCustomError={[MockFunction]}
        validationHook={
          Object {
            "name": "userEmail",
            "onBlur": [Function],
            "onChange": [Function],
            "ref": [Function],
          }
        }
      />
      <Segment
        basic={true}
        textAlign="center"
      >
        <Button
          as="button"
          content="      Next      "
          icon="right arrow"
          labelPosition="left"
          onClick={[Function]}
        />
      </Segment>
    </Form>
  </Card>
</Container>
`;
